src/components/controls/selects/MultiSelect/MultiSelect.tsx

```diff
@@ -0,0 +1,311 @@
+import { useDisclosure } from "@einride/hooks"
+import styled from "@emotion/styled"
+import { motion } from "framer-motion"
+import {
+  FocusEvent,
+  KeyboardEvent,
+  MouseEvent,
+  useEffect,
+  useId,
+  useMemo,
+  useRef,
+  useState,
+  ComponentPropsWithoutRef,
+} from "react"
+import { zIndex } from "../../../../lib/zIndex"
+import { Icon } from "../../../content/Icon/Icon"
+import { BoxProps, Box } from "../../../layout/Box/Box"
+// TODO move SearchSelect imports
+import { SearchSelectOption } from "../SearchSelect/SearchSelectOption"
+import { useScrollIntoView } from "./hooks/useScrollIntoView"
+import { useSelectedOptions } from "./hooks/useSelectedOptions"
+
+import { MultiSelectInput, MultiSelectInputSharedProps } from "./MultiSelectInput"
+import {
+  BaseOption,
+  MultiSelectWithLabelProps,
+  MultiSelectWithoutLabelProps,
+  Direction,
+} from "./types"
+
+interface MultiSelectBaseProps<Option> {
+  /** Props passed to dropdown element. */
+  dropdownProps?: ComponentPropsWithoutRef<"div"> & { "data-testid": string }
+
+  /** Filtering function to be used to populate dropdown. Filters on `option.value` by default. */
+  filter?: (value: string, option: Option) => boolean
+
+  /** If `false`, consumer have control over which options to pass to dropdown. Defaults to `true`. */
+  isFilterable?: boolean
+
+  /** Set if search term should be cleared after selecting an option. */
```

Comment:
Maybe something like this? I think it's nice to communicate the default when there is one. And "set" might be referring to both setting to true or false.

```suggestion
  /** If true, search term will be cleared after selecting an option. Default is `true`. */
```